arch/i386/linker.ld

```diff
@@ -1,30 +1,44 @@
-ENTRY(_start)
+
+ENTRY (_start)
+
 SECTIONS
 {
-	. = 1M;
+	kernel_virt_offset = 0xC0000000;
+	. = 0x00100000;
 
-	.text BLOCK(4K) : ALIGN(4K)
+	sk = .;
+
+	.multiboot.data : {
+		*(.multiboot.data)
+	}
+
+   .multiboot.text : {
+	   *(.multiboot.text)
+   }
+
+	.multiboot.bss : {
+		*(.multiboot.bss)
+	}
+
+	. += kernel_virt_offset;
+
+	.text ALIGN (4K) : AT (ADDR (.text) - 0xC0000000)
 	{
-		sk = .;
-		*(.multiboot)
 		*(.text)
 	}
```

Comment:
Inconsistencies in braces position: Either keep them at the end of the line or on their one but apply the same style for each section